processing/src/main/java/org/apache/druid/query/filter/DruidPredicateFactory.java

```diff
@@ -33,6 +38,11 @@ public interface DruidPredicateFactory
 
   DruidDoublePredicate makeDoublePredicate();
 
+  default Predicate<Object[]> makeArrayPredicate(@Nullable TypeSignature<ValueType> inputType)
+  {
+    throw new UOE("Predicate does not support ARRAY types");
```

Comment:
would be nice to do a `getClass()` or `toString()` or something here as the default method will just show up as the interface in the stack trace, not the concrete implementation.

Also, for these UOE, you can consider `DruidException.defensive()` or, if you are thinking that these should make it to an end-user, you could do something for the USER persona...

Reply:
Ah, `DruidException` has a `prependAndBuild()` method which takes the current DruidException and just prepends values to it.  The intent being that it can be used to add extra context from higher levels in the code.  

So, another option for that is to throw a `DruidException` here, catch it in whatever code might expect to see it and do the `prependAndBuild()` to throw with the added message.